CMakeLists.txt

```diff
@@ -18,10 +19,15 @@ add_library(convert_util SHARED convert_util.c)
 add_library(convert_client SHARED convert_client.c lib/log.c/src/log.c)
 target_link_libraries(convert_client PRIVATE syscall_intercept_static convert_util)
 
+# Unit & integration tests
+add_subdirectory(tests)
+
 # Tests for code linting & formatting
 enable_testing()
 
-set(LIBCONVERT_SOURCES convert_client.c convert.h convert_util.c convert_util.h)
+list(APPEND LIBCONVERT_SOURCES convert_client.c convert.h convert_util.c convert_util.h)
```

Comment:
(detail: why do you need a `list`? And why `APPEND`?)

Reply:
It actually already was a list before: set() creates a list if you pass multiple values.
Now I need to use the more-specific list() call to ask for appending. 

The cmake file in ./tests subdir will also add it own files to this list.
I did not want to duplicate the test logic for linting/code-formatting.
Instead, each sub-directory is in charge of appending the C files that should be cppcheck'ed/uncrustifi'ed to this list.

Reply:
OK I see. Maybe better to initialise the list here and append it in the subdirs, no?

Reply:
The problem is that list() does not support PARENT_SCOPE, so I have to use set() anyway from the subdir.
I use list(APPEND) deliberately in the root CMake file such that I don't have to worry about the order of execution.
There is probably a cleaner way to do that...

Reply:
I guess you could create a "function" to call cppcheck and uncrustify from the subdir. I don't know if it helps.
But it is fine for me to keep it like that

Reply:
(it is just that I don't know how you would manage the case were you have multiple subdirs)

Reply:
I am actually doing an append wit set() from the subdir:
set(LIBCONVERT_SOURCES ${LIBCONVERT_SOURCES} tests/check_convert_util.c PARENT_SCOPE)